lesson07/script.js

```diff
@@ -0,0 +1,44 @@
+const loading = document.getElementById('js-loading');
+
+function createLoading(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    loading.appendChild(loadingImg);
+}
+
+function removeLoading(){
+    loading.remove();
+}
+
+function createList(value){
+    const fragment = document.createDocumentFragment();
+    for(const list of value){
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = `/${list.to}`;
+        img.src = list.img;
+        img.alt = list.alt;
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    const ul = document.getElementById('js-list');
+    ul.appendChild(fragment);
+}
+
+function getList(){
+    return new Promise(resolve => {  
+    createLoading();
```

Comment:
IMO
I think createLoading should be outside of promise. I think the only value that should be resolved is lists.

```suggestion
   function getList(){
    createLoading();
    return new Promise(resolve => {  
```

Reply:
Thank you for your comment!
I certainly think so.
I didn't fully understand the challenges. I'll try fix it.